daffodil-tdml-processor/src/main/scala/org/apache/daffodil/tdml/processor/DaffodilTDMLDFDLProcessor.scala

```diff
@@ -274,6 +285,74 @@ class DaffodilTDMLDFDLProcessor private (private var dp: DataProcessor) extends
     new DaffodilTDMLUnparseResult(actual, outStream)
   }
 
+  def doParseWithBothApis(dpInputStream: java.io.InputStream, saxInputStream: java.io.InputStream,
+    lengthLimitInBits: Long): TDMLParseResult = {
+    val outputter = new TDMLInfosetOutputter()
+    outputter.setBlobAttributes(blobDir, blobPrefix, blobSuffix)
+
+    val xri = dp.newXMLReaderInstance
+    xri.setContentHandler(new SAXHandler())
+    xri.setErrorHandler(new DaffodilSAXErrorHandler())
+    xri.setProperty("BlobDirectory", blobDir)
+    xri.setProperty("BlobPrefix", blobPrefix)
+    xri.setProperty("BlobSuffix", blobSuffix)
+
+    val dis = InputSourceDataInputStream(dpInputStream)
+    val sis = InputSourceDataInputStream(saxInputStream)
+    if (lengthLimitInBits >= 0 && lengthLimitInBits % 8 != 0) {
+      // Only set the bit limit if the length is not a multiple of 8. In that
+      // case, we aren't expected to consume all the data and need a bitLimit
+      // to prevent messages about left over bits.
+      dis.setBitLimit0b(MaybeULong(lengthLimitInBits))
+      sis.setBitLimit0b(MaybeULong(lengthLimitInBits))
+    }
+    val actual = dp.parse(dis, outputter)
+    xri.parse(sis)
+    val errorHandler = xri.getErrorHandler.asInstanceOf[DaffodilSAXErrorHandler]
+
+    if( !actual.isError && !errorHandler.isError) {
+      verifySameParseOutput(outputter, xri)
+    }
+    verifySameDiagnostics(actual, xri)
+
+    new DaffodilTDMLParseResult(actual, outputter)
+  }
+
+  def verifySameParseOutput(dpOutputter: TDMLInfosetOutputter, xri: DFDL.DaffodilXMLReader): Unit = {
```

Comment:
Can this just accept the SAXHandler as the second param? I don't think it uses any other parts of the XMLReader.